sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -99,911 +53,859 @@ static az_span start_time_span;
-      pnp_process_twin_data(
-          &json_reader,
-          true,
-          sample_pnp_components,
-          sample_pnp_components_num,
-          sample_property_callback,
-          NULL);
+      LOG("Message Type: Desired Properties");
+      (void)pnp_process_device_twin_message(twin_message_span, true, pnp_components, pnp_components_num, property_callback, NULL);
       break;
 
-    // A response from a twin reported properties publish message. With a successful update of
-    // the reported properties, the payload will be empty and the status will be 204.
+    // A response from a twin reported properties publish message.
     case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
-      LOG_SUCCESS("A twin reported properties response message was received");
+      LOG("Message Type: Reported Properties");
       break;
   }
 }
 
-static az_result sample_pnp_temp_controller_process_command(
-    az_iot_hub_client_method_request* command_request,
-    az_span component_name,
-    az_span command_name,
-    az_span command_payload,
-    sample_pnp_mqtt_message* mqtt_message)
+static void handle_command_message(
+    az_span command_message_span,
+    const az_iot_hub_client_method_request* command_request)
 {
-  az_result result;
-
-  (void)command_payload;
-
-  if (az_span_ptr(component_name) == NULL
-      && az_span_is_content_equal(reboot_command_name, command_name))
+  az_result rc;
+  az_span command_name;
+  az_span component_name;
+  az_span command_response_payload;
+  az_iot_status status = AZ_IOT_STATUS_UNKNOWN;
+
+  pnp_parse_command_name(command_request->name, &component_name, &command_name);
+
+  // Invoke command and retrieve response to send to server.
+  if (az_succeeded(pnp_thermostat_process_command(
+          &hub_client,
+          &thermostat_1,
+          command_request,
+          component_name,
+          command_name,
+          command_message_span,
+          &mqtt_message,
+          &status)))
+  {
+    LOG_AZ_SPAN("Client invoked command on Temperature Sensor 1:", command_name);
+    command_response_payload = mqtt_message.out_payload_span;
+  }
+  else if (az_succeeded(pnp_thermostat_process_command(
+               &hub_client,
+               &thermostat_2,
+               command_request,
+               component_name,
+               command_name,
+               command_message_span,
+               &mqtt_message,
+               &status)))
   {
-    // This is a command for the device
+    LOG_AZ_SPAN("Client invoked command on Temperature Sensor 2:", command_name);
+    command_response_payload = mqtt_message.out_payload_span;
+  }
+  else if (az_succeeded(temp_controller_process_command(
+               command_request,
+               component_name,
+               command_name,
+               command_message_span,
+               &mqtt_message,
+               &status)))
+  {
+    LOG_AZ_SPAN("Client invoked command on controller:", command_name);
```

Comment:
nit:
```suggestion
    LOG_AZ_SPAN("Client invoked command on temperature controller:", command_name);
```
only because I initially interpreted this as "microcontroller"